src/lib/supabase.ts

```diff
@@ -36,31 +31,75 @@ export async function authenticatedVote (user_id: number, repo_id: number) {
 
   await supabase
     .from("users_to_repos_votes")
-    .update({ deleted_at: (new Date).toISOString() })
+    .update({ deleted_at: new Date().toISOString() })
     .eq("user_id", user_id)
     .eq("repo_id", repo_id);
 
   return -1;
 }
 
-export async function updateVotesByRepo (votes: number, repo_id: number, user_id: number) {
+export async function authenticatedStar(user_id: number, repo_id: number) {
```

Comment:
this is not correct, we would need to support the API here, not supabase: https://beta.api.opensauced.pizza/#/Repository%20service%20guarded/starOneById

Reply:
Can you tell me more about this?
I tried to use `SWR` but can't call inside a function. So I'm thinking something like this:
```js
// hooks/useStarRepos.ts
const useStarRepos = (repo_id: number) => {
  const { data, error, mutate } = useSWR<PaginatedRepoResponse, Error>(`repos/${repo_id}/star`);
...
```

Reply:
You can't SWR a POST request, can just use fetch for that; however we should wait for @bdougie to make sure we got the feature logic right - it might be that the API design needs a little bit of polishing 😓

Reply:
@0-vortex any updates on this?

Reply:
Not sure what context is missing and where I can help, but the idea is to start the repo without going to GItHub here. 
![Screen Shot 2022-11-29 at 3 40 16 PM](https://user-images.githubusercontent.com/5713670/204672454-a530c17e-f93d-4136-abad-a053ccc3aa8a.png)

Reply:
I used `supabase` to do that (not working for the search results yet, only for hot repos and others) but it seems like I should use the API: [https://beta.api.opensauced.pizza/v1/repos/{id}/star](https://beta.api.opensauced.pizza/#/Repository%20service%20guarded/starOneById)

Reply:
update:
it works for the search results :ok_hand:

Reply:
I am not seeing it working when trying to start facebook/create-react-app. Any chance you can record a Loom to show it on your end?

 
<img width="462" alt="Screen Shot 2022-12-01 at 4 56 21 AM" src="https://user-images.githubusercontent.com/5713670/205058548-bcb018ba-fdb7-493b-a4dd-7fa33c5ac644.png">

Reply:
@bdougie https://www.loom.com/share/44542baa5e3344e596782781ab063665

Reply:
there is a little bit of delay though, maybe I should add a loading indicator and disable it :thinking:

Reply:
Ok, I can confirm that works for me, just not in the search results (my screenshot above).

![Screen Shot 2022-12-01 at 10 22 20 AM](https://user-images.githubusercontent.com/5713670/205130656-c3ec4f50-4963-4afe-9e97-5f0d8309ce1c.png)

I am not sure the API is starting the repo though. Perhaps it doesn't have access. Can you confirm that the repo you clicked is starred on GitHub. If not, we should open an issue here - https://github.com/open-sauced/api.opensauced.pizza

![Screen Shot 2022-12-01 at 10 23 46 AM](https://user-images.githubusercontent.com/5713670/205130915-2933a29e-50b1-466c-9751-d3515e4e3a6c.png)

I can confirm that I am not starred on GitHub.

Reply:
No, it's not starred on Github, IMO we are only saving the information in Supabase.
@0-vortex is that correct?

:information_source:  I'm not using the [api.opensauced.pizza](https://beta.api.opensauced.pizza/) to star a repo. I did it the same way the `vote` was done.

I definitely need more information about how to use it :arrow_down: 

> this is not correct, we would need to support the API here, not supabase: https://beta.api.opensauced.pizza/#/Repository%20service%20guarded/starOneById
> however we should wait for @bdougie to make sure we got the feature logic right - it might be that the API design needs a little bit of polishing sweat
>
[:arrow_up:  comment](https://github.com/open-sauced/hot/pull/387#discussion_r1004505626)

Reply:
I looked at the video again and I see the search box works for you as well. Would be beneficial to get @open-sauced/triage team repo back on their experiences.

Reply:
> IMO we are only saving the information in Supabase.
 
I think that would be confusing because after looking at the video I assumed as well that clicking the star was starring the repo as well.

Reply:
> IMO we are only saving the information in Supabase.
 
I think that would be confusing because after looking at the video I assumed as well that clicking the star was starring the repo.

Reply:
Previously asked about logic confirmation but have to push back on this because: since the inception of "user_stars" this has been the list of repositories loaded by open-sauced users into app.opensauced.pizza - moving this feature into SupaBase it was shortened to "stars" meaning "opensauced stars" and then, later on, when the UI was rehauled to display icons, less icons were implemented than the number of features we had (4 not counting GitHub data), using the same icon for GitHub and open-sauced stars feature.

However, there are some limitations we can not overcome like:
- we are already hitting the GitHub client rate limit across one GitHub app and 3 clients, we can't afford to wrap other GitHub actions in the hot endpoint
- we are using GitHub login as auth layer only and don't have the GitHub user stars information available right now

For the purpose of this pull request, replacing the SupaBase client functions and code with calls to the backend is enough. If the stars feature needs to be renamed in the frontend and backend, super happy to do that, if not but we also want open sauced stars to mimic GitHub stars, we would have to explore what that means and how we change current features, from counting open sauced stars (repos to stars table right now) to how we display them (what open sauced users starred what repo) since none of that is possible or has been planned for at the moment; I did mention this idea in the open source sync but it was on an explorative basis, to spawn the tracking and fragment issues needed to do this.

@bdougie it looks to me like the stars and vote feature on the api level should be merged into "vote" and displaying any stars data should be mimicking GitHub information. Is that correct? If so, we should have a talk/brainstorming session about this 😅

Reply:
Ok, I think there is confusion across the board. The star feature was included in the design from last summer and a nice to have but not required. 

I agree we should focus on voting and remove stars altogether from the API. We can display stars and punt on trying to make stars on GitHub work (which turns out was not the focus here).

Thanks for your work on this @TAKANOME-DEV, but I think this exploration have proven to not be needed.

Reply:
I totally agree, thank you for taking the time to go over this PR :pray: